### ISSUE TITLE
Add database unit tests

### ISSUE THREAD

**Issue by vitaly-t** [LiskHQ/lisk#1484]
Changes within the database layer during the last month invalidate the existing approach to testing database-related functionality.

The updated database platform allows carrying out tests in complete isolation for each of the database repositories that's now available.

A new folder and structure is required within the `/test/db`, to contain files like `blocks.js`, `accounts.js`, etc, i.e. exactly matching the set of files of the corresponding database repository to be tested, as they can be found in folder `/db/repos`.

Once all the tests related to the use of database repositories are done like this, we can start simplifying core tests, to focus only on the business logic of the application.

---

## Unit tests 
- [x] accounts #1537
- [x] blocks #1565
- [x] dapps #1566
- [x] delegates #1567
- [x] migrations #1589 
- [x] multisignatures #1590 
- [x] peers #1591 
- [x] rounds #1592 
- [x] transactions #1593 
- [x] transactions.* #1594 
- [x] voters #1595 
- [x] votes #1596

**Comment by chiptus** [comment]
can I have a take on this?

**Comment by vitaly-t** [comment]
@chiptus It is already a work-in-progress: https://github.com/LiskHQ/lisk/tree/1484-database-tests/test/unit/db.

And I'm planning to get back to it this evening.

**Comment by nazarhussain** [comment]
@vitaly-t  @karmacoma  I updated the issue description and removed these tasks from list. 

 - Create a new `/test/db` folder + file structure, replicating that of the database repositories
 - Implement a generic stub interface for each repository to be tested in compete isolation 
 - Create one sample, showing the use of the repository stub for testing

As the first point is covered. And for rest we decided to not mock DB I/O interface.
